library/CMakeLists.txt

```diff
@@ -408,8 +408,8 @@ endif()
 target_link_libraries(dfhack protobuf-lite clsocket lua jsoncpp_static dfhack-version ${PROJECT_LIBS})
 set_target_properties(dfhack PROPERTIES INTERFACE_LINK_LIBRARIES "")
 
-target_link_libraries(dfhack-client protobuf-lite clsocket jsoncpp_static)
-target_link_libraries(dfhack-run dfhack-client)
+target_link_libraries(dfhack-client protobuf-lite clsocket jsoncpp_static dfhack)
+target_link_libraries(dfhack-run dfhack-client dfhack)
```

Comment:
Coming from https://github.com/DFHack/dfhack/pull/3524#issuecomment-1620621678... why's the link-time dependency necessary now? As far as I can tell, the only change to RemoteClient is that it pulls in the `command_result` enum from Core.h now, which shouldn't introduce a link-time dependency. Attempting to call any functions from Core.h would introduce such a dependency, but we can avoid that.

Reply:
yeah, I wasn't clear on why this dependency showed up either. let me look into this a little more.

Reply:
the link error only occurs on Windows:
```gcc
FAILED: library/dfhack-client.dll library/dfhack-client.lib 
: && /opt/cmake/bin/cmake -E vs_link_dll --intdir=library/CMakeFiles/dfhack-client.dir --rc=/opt/msvc/bin/x64/rc --mt=/opt/msvc/bin/x64/mt --manifests  -- /opt/msvc/bin/x64/link /nologo library/CMakeFiles/dfhack-client.dir/RemoteClient.cpp.obj library/CMakeFiles/dfhack-client.dir/ColorText.cpp.obj library/CMakeFiles/dfhack-client.dir/MiscUtils.cpp.obj library/CMakeFiles/dfhack-client.dir/Error.cpp.obj library/CMakeFiles/dfhack-client.dir/proto/Basic.pb.cc.obj library/CMakeFiles/dfhack-client.dir/proto/BasicApi.pb.cc.obj library/CMakeFiles/dfhack-client.dir/proto/CoreProtocol.pb.cc.obj library/CMakeFiles/dfhack-client.dir/Console-windows.cpp.obj  /out:library/dfhack-client.dll /implib:library/dfhack-client.lib /pdb:library/dfhack-client.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO  depends/protobuf/protobuf-lite.lib  depends/clsocket/clsocket.lib  depends/jsoncpp-sub/src/lib_json/jsoncpp.lib  depends/zlib/lib/zlib.lib  Ws2_32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && :
LINK: command "/opt/msvc/bin/x64/link /nologo library/CMakeFiles/dfhack-client.dir/RemoteClient.cpp.obj library/CMakeFiles/dfhack-client.dir/ColorText.cpp.obj library/CMakeFiles/dfhack-client.dir/MiscUtils.cpp.obj library/CMakeFiles/dfhack-client.dir/Error.cpp.obj library/CMakeFiles/dfhack-client.dir/proto/Basic.pb.cc.obj library/CMakeFiles/dfhack-client.dir/proto/BasicApi.pb.cc.obj library/CMakeFiles/dfhack-client.dir/proto/CoreProtocol.pb.cc.obj library/CMakeFiles/dfhack-client.dir/Console-windows.cpp.obj /out:library/dfhack-client.dll /implib:library/dfhack-client.lib /pdb:library/dfhack-client.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO depends/protobuf/protobuf-lite.lib depends/clsocket/clsocket.lib depends/jsoncpp-sub/src/lib_json/jsoncpp.lib depends/zlib/lib/zlib.lib Ws2_32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:library/dfhack-client.dll.manifest" failed (exit code 96) with the following output:
   Creating library library/dfhack-client.lib and object library/dfhack-client.exp
RemoteClient.cpp.obj : error LNK2019: unresolved external symbol "private: __cdecl DFHack::Core::Core(void) __ptr64" (??0Core@DFHack@@AEAA@XZ) referenced in function "public: static class DFHack::Core & __ptr64 __cdecl DFHack::Core::getInstance(void)" (?getInstance@Core@DFHack@@SAAEAV12@XZ)
RemoteClient.cpp.obj : error LNK2019: unresolved external symbol "private: __cdecl DFHack::Core::~Core(void) __ptr64" (??1Core@DFHack@@AEAA@XZ) referenced in function "??__Finstance@?1??getInstance@Core@DFHack@@SAAEAV12@XZ@YAXXZ" (??__Finstance@?1??getInstance@Core@DFHack@@SAAEAV12@XZ@YAXXZ)
library\dfhack-client.dll : fatal error LNK1120: 2 unresolved externals
```

because this is in the header file:
```cpp
        static Core& getInstance()
        {
            static Core instance;
            return instance;
        }
```

I don't see why this is in the header file and not the cpp file. I'll try to move it.